pixeltable/functions/gemini.py

```diff
@@ -242,6 +247,118 @@ def _(model: str) -> str:
     return f'request-rate:veo:{model}'
 
 
+# The vector dimensionality limit imposed by pgvector (note: Pixeltable does not use halfvec yet)
+_MAX_EMBEDDING_DIMENSIONALITY = 2000
+# The highest embedding dimensionality recommended by Google that is within the above limit
+_DEFAULT_EMBEDDING_DIMENSIONALITY = 1536
+
+
+@pxt.udf(resource_pool='request-rate:gemini', batch_size=32)
+async def generate_embedding(
+    input: Batch[str], *, model: str = 'gemini-embedding-001', config: dict | None = None, async_: bool = False
+) -> Batch[pxt.Array[(None,), pxt.Float]]:
+    """Generate embeddings for the input strings. For more information on Gemini embeddings API, see:
+    https://ai.google.dev/gemini-api/docs/embeddings
+
+    This UDF can run in either synchronous or asynchronous mode. Synchronous mode is optimized for latency.
+    Asynchronous mode uses Gemini's Batch API described at https://ai.google.dev/gemini-api/docs/batch-api which
+    provides a better throughput at lower cost at the expense of higher latency.
+
+    The `output_dimensionality` in the config must not exceed 2000, if present.
+
+    __Requirements:__
+
+    - `pip install google-genai`
+
+    Args:
+        input: The strings to generate embeddings for.
+        model: The Gemini model to use.
+        config: Configuration for embedding generation, corresponding to keyword arguments of
+            `genai.types.EmbedContentConfig`. For details on the parameters, see:
+            <https://googleapis.github.io/python-genai/genai.html#genai.types.EmbedContentConfig>
+        async_: whether asynchronous mode should be used.
+
+    Returns:
+        The generated embeddings.
+
+    Examples:
+        Add a computed column with embeddings to an existing table with a `text` column:
+
+        >>> t.add_computed_column(embedding=generate_embedding(t.text))
+
+        Add an embedding index on `text` column:
+
+        >>> t.add_embedding_index(t.text, embedding=generate_embedding.using(model='gemini-embedding-001', async_=True))
```

Comment:
I find the inclusion of the `async_` parameter confusing here, especially with no additional context given. Unsuspecting users might presume it's somehow related to the fact that it's an embedding index, since that's the only noted difference with the preceding example. Clearest to just leave it out.